language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/OnConflictClauseNodeContext.java

```diff
@@ -57,12 +65,37 @@ public OnConflictClauseNodeContext() {
             completionItems.addAll(this.expressionCompletions(context));
         }
         this.sort(context, node, completionItems);
-        
+
         return completionItems;
     }
 
     @Override
     public boolean onPreValidation(BallerinaCompletionContext context, OnConflictClauseNode node) {
         return !node.onKeyword().isMissing() && !node.conflictKeyword().isMissing();
     }
+
+    @Override
+    public void sort(BallerinaCompletionContext context, OnConflictClauseNode node, List<LSCompletionItem> lsCItems) {
+        for (LSCompletionItem lsCItem : lsCItems) {
+            int rank = 2;
+            if (this.isVariableOfErrorType(lsCItem)) {
+                Optional<Symbol> symbol = ((SymbolCompletionItem) lsCItem).getSymbol();
+                Optional<TypeSymbol> typeDescriptor = SymbolUtil.getTypeDescriptor(symbol.get());
+                if (typeDescriptor.isPresent()
+                        && CommonUtil.getRawType(typeDescriptor.get()).typeKind() == TypeDescKind.ERROR) {
+                    rank = 1;
+                }
+            }
+            String sortText = SortingUtil.genSortText(rank)
+                    + SortingUtil.genSortText(SortingUtil.toRank(context, lsCItem));
+            lsCItem.getCompletionItem().setSortText(sortText);
+        }
+    }
+
+    private boolean isVariableOfErrorType(LSCompletionItem lsCItem) {
+        Predicate<Symbol> symbolPredicate = CommonUtil.getVariableFilterPredicate();
```

Comment:
How about function call expressions of error type? We haven't considered them here.

Reply:
Done